Cabal/src/Distribution/Simple/GHC.hs

```diff
@@ -717,7 +717,7 @@ buildOrReplLib mReplFlags verbosity numJobs pkg_descr lbi lib clbi = do
             , ghcOptFPic = toFlag True
             , ghcOptHiSuffix = toFlag "dyn_hi"
             , ghcOptObjSuffix = toFlag "dyn_o"
-            , ghcOptExtra = hcSharedOptions GHC libBi
+            , ghcOptExtra = hcOptions GHC libBi ++ hcSharedOptions GHC libBi
```

Comment:
For consistency shouldn't you do the same thing for `ghcOptExtra` of `profOpts` a few line above?